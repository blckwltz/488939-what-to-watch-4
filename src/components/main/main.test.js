import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main';

const mocks = {
  featuredMovieTitle: `The Grand Budapest Hotel`,
  featuredMovieGenre: `Drama`,
  featuredMovieReleaseDate: 2014,
  movieTitles: [`Fantastic Beasts`, `Bohemian Rhapsody`, `Macbeth`],
};
const onTitleClick = () => {};

it(`Should render correctly`, () => {
  const tree = renderer
    .create(
        <Main
          featuredMovieTitle={mocks.featuredMovieTitle}
          featuredMovieGenre={mocks.featuredMovieGenre}
          featuredMovieReleaseDate={mocks.featuredMovieReleaseDate}
          movieTitles={mocks.movieTitles}
          onTitleClick={onTitleClick}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
