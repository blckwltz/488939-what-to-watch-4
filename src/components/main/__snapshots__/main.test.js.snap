// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Main component correctly 1`] = `
Array [
  <section
    className="movie-card"
    style={
      Object {
        "background": "#ffffff",
      }
    }
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Pulp Fiction"
        src="img/the-grand-budapest-hotel-poster.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="/img/avatar.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Pulp Fiction"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Pulp Fiction
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              1994
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Adventure
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Biography
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Mystery
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/1"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/2"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/3"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/4"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              Shutter Island
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/5"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              Snatch
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/6"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Pulp Fiction
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/7"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <a
              href="/films/8"
              onClick={[Function]}
            >
              <video
                height="175"
                width="280"
              />
            </a>
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/8"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
